services/vision/mlvision/classifier.go

```diff
@@ -45,9 +45,12 @@ func attemptToBuildClassifier(mlm mlmodel.Service) (classification.Classifier, e
 			return nil, err
 		}
 
-		probs := unpack(outMap, "probability")
-		if len(probs) == 0 {
-			probs = unpack(outMap, DefaultOutTensorName+"0")
+		probs, err := unpack(outMap, "probability")
+		if err != nil || len(probs) == 0 {
+			probs, err = unpack(outMap, DefaultOutTensorName+"0")
+			if err != nil {
+				return nil, err
```

Comment:
Rather than overwrite err, can you combine the two errors? You can use multierr.Combine
https://pkg.go.dev/go.uber.org/multierr#section-readme
you will need to define err2 outside of the scope of the if statement in order for "probability" to be correctly filled, too.